src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -81,4 +84,12 @@ public void storeMessage(Message message) {
 
     return messages;
   }
+
+  /** Returns the total number of messages for all users. */
+  public int getTotalMessageCount(){
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
+    return results.countEntities(FetchOptions.Builder.withLimit(1000));
```

Comment:
It looks like this limit would make it so that the app would always return "1000" if it had very high usage. While it may make sense to specify the "chunkSize", you probably do not want a fixed limit so that you actually give the total count.